test/unit/test-action-allowlist.js

```diff
@@ -0,0 +1,446 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../../extensions/amp-carousel/0.1/amp-carousel';
+import '../../extensions/amp-image-lightbox/0.1/amp-image-lightbox';
+import '../../extensions/amp-lightbox/0.1/amp-lightbox';
+import '../../extensions/amp-list/0.1/amp-list';
+import '../../extensions/amp-selector/0.1/amp-selector';
+import '../../extensions/amp-sidebar/0.1/amp-sidebar';
+import {ActionInvocation, ActionService} from '../../src/service/action-impl';
+import {ActionTrust} from '../../src/action-constants';
+import {AmpForm} from '../../extensions/amp-form/0.1/amp-form';
+import {Services} from '../../src/services';
+import {
+  createElementWithAttributes,
+  whenUpgradedToCustomElement,
+} from '../../src/dom';
+import {whenCalled} from '../../testing/test-helper.js';
+
+function createExecElement(id, enqueAction, defaultActionAlias) {
+  const execElement = document.createElement('amp-element');
+  execElement.setAttribute('id', id);
+  return setExecElement(execElement, enqueAction, defaultActionAlias);
+}
+
+function setExecElement(element, enqueAction, defaultActionAlias) {
+  element.enqueAction = enqueAction;
+  element.getDefaultActionAlias = defaultActionAlias;
+  return element;
+}
+
+function getActionInvocation(element, action, opt_tagOrTarget) {
+  return new ActionInvocation(
+    element,
+    action,
+    /* args */ null,
+    'source',
+    'caller',
+    'event',
+    ActionTrust.HIGH,
+    'tap',
+    opt_tagOrTarget || element.tagName
+  );
+}
+
+describes.realWin(
+  'Action allowlist on components',
+  {
+    amp: {
+      runtimeOn: true,
+      extensions: [
+        'amp-carousel',
+        'amp-form',
+        'amp-image-lightbox',
+        'amp-lightbox',
+        'amp-list',
+        'amp-selector',
+        'amp-sidebar',
```

Comment:
Feels like an integration test if we're relying on extensions to be built and run but we're inspecting private state in the test code below.

Reply:
This particular test case is in more of a grey area, since it is not testing the interaction between any two components per se, but each component against the action service. Perhaps as a compromise we could move each case to that component's corresponding unit tests? This way the action-specific unit tests are not crowded with building many extensions.

Reply:
Yea one actual issue is that tests in `test/unit` may be run before extensions are built in Travis. Moving to unit tests for each component sounds good.

Alternatively, you can make this file an integration test but I'd strongly recommend against verifying private state if you do. Generally this is a bad idea even in unit tests but alas.